src/Temporalio/Worker/Tuning/WorkerTuner.cs

```diff
@@ -0,0 +1,106 @@
+            ResourceBasedSlotSupplierOptions? localActivityOptions = null)
+        {
+            ResourceBasedTunerOptions tunerOpts =
+                new ResourceBasedTunerOptions(targetMemoryUsage, targetCpuUsage);
+            return new CompositeTuner(
+                new ResourceBasedSlotSupplier(
+                    workflowOptions == null
+                        ? new ResourceBasedSlotSupplierOptions()
+                        : workflowOptions,
+                    tunerOpts),
+                new ResourceBasedSlotSupplier(
+                    activityOptions == null
+                        ? new ResourceBasedSlotSupplierOptions()
+                        : activityOptions,
+                    tunerOpts),
+                new ResourceBasedSlotSupplier(
+                    localActivityOptions == null
+                        ? new ResourceBasedSlotSupplierOptions()
+                        : localActivityOptions,
+                    tunerOpts));
+        }
+
+        /// <summary>
+        /// Create a fixed-size tuner with the given slot capacities.
+        /// </summary>
+        /// <param name="numWorkflowTaskSlots">The number of available workflow task slots.</param>
+        /// <param name="numActivityTaskSlots">The number of available activity task slots.</param>
+        /// <param name="numLocalActivitySlots">The number of available local activity slots.</param>
+        /// <returns>The tuner.</returns>
+        public static WorkerTuner CreateFixed(
+            uint numWorkflowTaskSlots,
+            uint numActivityTaskSlots,
+            uint numLocalActivitySlots)
+        {
+            return new CompositeTuner(
+                new FixedSizeSlotSupplier(numWorkflowTaskSlots),
+                new FixedSizeSlotSupplier(numActivityTaskSlots),
+                new FixedSizeSlotSupplier(numLocalActivitySlots));
+        }
+
+        /// <summary>
+        /// Create a tuner composed of the provided slot suppliers.
+        /// </summary>
+        /// <param name="workflowTaskSlotSupplier">The supplier of workflow task slots.</param>
+        /// <param name="activityTaskSlotSupplier">The supplier of activity task slots.</param>
+        /// <param name="localActivitySlotSupplier">The supplier of local activity slots.</param>
+        /// <returns>The tuner.</returns>
+        public static WorkerTuner CreateComposite(
+            ISlotSupplier workflowTaskSlotSupplier,
+            ISlotSupplier activityTaskSlotSupplier,
+            ISlotSupplier localActivitySlotSupplier)
+        {
+            return new CompositeTuner(
+                workflowTaskSlotSupplier,
+                activityTaskSlotSupplier,
+                localActivitySlotSupplier);
+        }
+
+        /// <summary>
+        /// Gets a slot supplier for workflow tasks.
+        /// </summary>
+        /// <returns>A slot supplier for workflow tasks.</returns>
+        public abstract ISlotSupplier GetWorkflowTaskSlotSupplier();
```

Comment:
```suggestion
        public abstract ISlotSupplier WorkflowTaskSlotSupplier { get; }
```

Lets make these abstract properties instead of getters